src/Sylius/Bundle/CoreBundle/spec/Checkout/CheckoutProcessScenarioSpec.php

```diff
@@ -39,20 +39,25 @@ function it_implements_Sylius_process_scenario_interface()
         $this->shouldImplement('Sylius\Bundle\FlowBundle\Process\Scenario\ProcessScenarioInterface');
     }
 
-    function it_builds_checkout_process_with_proper_steps(ProcessBuilderInterface $builder)
+    function it_builds_checkout_process_with_proper_steps(ProcessBuilderInterface $builder, CartInterface $cart)
     {
-        $builder->add('security', 'sylius_checkout_security')->willReturn($builder)->shouldBeCalled();
-        $builder->add('addressing', 'sylius_checkout_addressing')->willReturn($builder)->shouldBeCalled();
-        $builder->add('shipping', 'sylius_checkout_shipping')->willReturn($builder)->shouldBeCalled();
-        $builder->add('payment', 'sylius_checkout_payment')->willReturn($builder)->shouldBeCalled();
-        $builder->add('finalize', 'sylius_checkout_finalize')->willReturn($builder)->shouldBeCalled();
-        $builder->add("purchase", "sylius_checkout_purchase")->willReturn($builder)->shouldBeCalled();
-
-        $builder->setDisplayRoute('sylius_checkout_display')->willReturn($builder)->shouldBeCalled();
-        $builder->setForwardRoute('sylius_checkout_forward')->willReturn($builder)->shouldBeCalled();
-
-        $builder->setRedirect('sylius_homepage')->willReturn($builder)->shouldBeCalled();
-        $builder->validate(Argument::any())->willReturn($builder)->shouldBeCalled();
+        $cart->getNumber()->willReturn('000000001');
+
+        $builder->add('security', 'sylius_checkout_security')->willReturn($builder);
+        $builder->add('addressing', 'sylius_checkout_addressing')->willReturn($builder);
+        $builder->add('shipping', 'sylius_checkout_shipping')->willReturn($builder);
+        $builder->add('payment', 'sylius_checkout_payment')->willReturn($builder);
+        $builder->add('finalize', 'sylius_checkout_finalize')->willReturn($builder);
+        $builder->add("purchase", "sylius_checkout_purchase")->willReturn($builder);
+
+        $builder->setDisplayRoute('sylius_checkout_display')->willReturn($builder);
+        $builder->setForwardRoute('sylius_checkout_forward')->willReturn($builder);
+
+        $builder->setRedirect('sylius_account_order_show')->willReturn($builder);
+        $builder->setRedirectParams(array(
+            'number' => '000000001'
+        ))->willReturn($builder);
+        $builder->validate(Argument::any())->willReturn($builder);
```

Comment:
I agree that we should build specs in such a way that we use either `mock` or `stub`, but with the builder's fluent interface it's not possible. Without `shouldBeCalled` there is no assertion, so the spec will pass everytime, even when the `build` method will be empty. Why we want to have spec that doesn't validate anything?